Setup/Patch/Data/CustomCustomerAttributeSyncedToYotpoCustomer.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Yotpo\SmsBump\Setup\Patch\Data;
+
+use Yotpo\Core\Model\CustomCustomerAttributeSyncedToYotpoCustomer as CoreCustomCustomerAttributeSyncedToYotpoCustomer;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Customer\Setup\CustomerSetupFactory;
+use Magento\Customer\Model\Customer;
+
+/**
+ * Class CustomCustomerAttributeSyncedToYotpoCustomer
+ * Add custom attribute to customer
+ */
+// phpcs:ignore
+class CustomCustomerAttributeSyncedToYotpoCustomer extends CoreCustomCustomerAttributeSyncedToYotpoCustomer implements DataPatchInterface
+{
+    /**
+     * @var ModuleDataSetupInterface
+     */
+    private $moduleDataSetup;
+
+    /**
+     * @var CustomerSetupFactory
+     */
+    private $customerSetupFactory;
+
+    /**
+     * CustomCustomerAttributeSmsMarketing constructor.
+     * @param ModuleDataSetupInterface $moduleDataSetup
+     * @param CustomerSetupFactory $customerSetupFactory
+     */
+    public function __construct(
+        ModuleDataSetupInterface $moduleDataSetup,
+        CustomerSetupFactory $customerSetupFactory
+    ) {
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->customerSetupFactory = $customerSetupFactory;
+    }
+
+    /**
+     * Creates custom attribute - synced_to_yotpo_customer
+     *
+     * @return void|CustomCustomerAttributeSyncedToYotpoCustomer
+     * @throws LocalizedException
+     * @throws \Zend_Validate_Exception
+     */
+    public function apply()
+    {
+        $customerSetup = $this->customerSetupFactory->create(['setup' => $this->moduleDataSetup]);
+        $customerSetup->addAttribute(Customer::ENTITY, 'synced_to_yotpo_customer', [
+            'type' => 'int',
+            'required' => false,
+            'visible' => false,
+            'user_defined' => false,
+            'label' => 'Synced to yotpo customer',
```

Comment:
1. There is no scope definition applicable for customer attributes in magento. In other words al attributes are global, as there is no store specific attribute management in place
2. The sync observer is not working, if we set a default value here. When a new customer is created, the real time sync works as part of observer and it updates the attribute value to 1. But finally it sets the attribute flag to 0 even though the sync was success. To overcome this issue, we shouldn't set a default value to 0
3. user_defined=false attributes can't be deleted from magento admin. As these attributes are necessary for our extension to work, we have put these like that to avoid any accidental deletion of such attributes by an admin.

Reply:
Can you please elaborate about No. 2?
Please point me to the relevant section in the code and provide more details why would a default value cause issues with the sync.

Reply:
Scenario:

1. Create a new customer
2. The customer syncs to Yotpo as part of realtime sync
3. This time the attribute value should be set as 1 to indicate that the customer has been synced to yotpo so that it shouldn't be picked up by the cron next time
4. When we set the default value as 0, the attribute value is set back to 0 after there is a sync.

Yotpo\SmsBump\Observer::execute() is the observer that gets triggered when a new customer account is created. After this observer is executed, the code will flow back to core magento classes and the default value is saved back somewhere there by simply ignoring what we set as part of our observer and the sync status. As an impact the same customer record will be synced one more time as part of the cron job. 

To get the exact file lines and details, we will require more time to debug into the core magento code.  

But please note that there isn't any benefit in setting a default value. It won't have any effect on the historical customer records . Also we are correctly setting a value for  the attribute (1 Or 0) depending upon the status of real-time sync. As  it causes an issue there on new customer registration, we just didn't set a default value there. Everything then worked as expected.